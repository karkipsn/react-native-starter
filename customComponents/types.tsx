import { 
    Text as DefaultText, 
    View as DefaultView ,
    TouchableOpacityProps,
    ImageSourcePropType
} from 'react-native';

// Themeprops for dark and white theme
export type ThemeProps = {
    lightColor?: string;
    darkColor?: string;
  };

// Adding themeprops of lightcolor and darkColor with the default Props of Text and View

export type TextProps = ThemeProps & DefaultText['props'];
export type ViewProps = ThemeProps & DefaultView['props'];


// Custom Button Props with Pressable Component and own props
export type CustomButtonProps = {
    title: string,
    radius: number,
    onPress?: () => void;
}

// custom Button elwement with the TouchableOpacity for IOS and TOuchableNativeFeedback for Android
export interface CustomTouchableOpacityProps extends TouchableOpacityProps {
    textTitle: string,
    backgroundColor: string
    radius: number
}


// Row for dispalying the contact data the contact Screen.
export type RowProps = {
    image : ImageSourcePropType,
    title: string,
    subtitle: string,
    onPress?: () => void;
}


export type SimpleRowProps = {
  title?: string,
  subtitle?: string
}

