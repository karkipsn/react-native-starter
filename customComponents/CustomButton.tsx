
import {CustomTouchableOpacityProps, 
    CustomButtonProps } from './types';

import * as React from 'react';

import { View, 
    StyleSheet, 
    Platform,
    TouchableOpacity, 
    TouchableNativeFeedback,
    Text, 
    Pressable, 
    TouchableOpacityProps } from "react-native";


/* Using Pressable */
export const Button = (props: CustomButtonProps) => {

    return (
      <Pressable style={[styles.button, {borderRadius: props.radius}]} onPress={props.onPress}>
        <Text style={styles.text}>{props.title}</Text>
      </Pressable>
    );
  }

  /* Using TouichableOpacity and merging the new required props to the props of the */
  export const CustomButton = (props: CustomTouchableOpacityProps) => {

    if (Platform.OS == 'ios') {

        return (
            <View style={styles.container}>
                    
              <TouchableOpacity {...props} onPress={props.onPress} style={[ styles.button, 
              {borderRadius: props.radius? props.radius: 20, width: "90%", backgroundColor: props.backgroundColor } ]} >
          
                  <Text style={styles.customBtnText}>{props.textTitle}</Text>
          
              </TouchableOpacity>
              </View>
              )
    } else {
        return (
            <View style={styles.container}>
                    
              <TouchableNativeFeedback {...props} onPress={props.onPress} style={[ styles.button, 
              {borderRadius: props.radius? props.radius: 20, width: "90%", backgroundColor: props.backgroundColor } ]} >
          
                  <Text style={styles.customBtnText}>{props.textTitle}</Text>
          
              </TouchableNativeFeedback>
              </View>
              )
    }

  }

// these styles are taken as default values
const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center"
    },

    customBtnBG: {
        backgroundColor: "#007aff",
        paddingHorizontal: 30,
        paddingVertical: 5,
        borderRadius: 30
    },
  
    customBtnText: {
        fontSize: 40,
        fontWeight: '400',
        color: "#fff",
    },
    button: {
        alignItems: 'center',
        justifyContent: 'center',
        paddingVertical: 12,
        paddingHorizontal: 32,
        borderRadius: 4,
        elevation: 3,
        backgroundColor: 'red',
      },
      text: {
        fontSize: 16,
        lineHeight: 21,
        fontWeight: 'bold',
        letterSpacing: 0.25,
        color: 'white',
      },
  });

  export default {CustomButton, Button};