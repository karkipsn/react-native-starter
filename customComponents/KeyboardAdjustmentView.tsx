import React, { ReactElement } from "react";
import {KeyboardAvoidingView, ScrollView, Platform, ViewStyle} from 'react-native';


export type Props = {
    style?: ViewStyle;
}

const KeyboardAdjustmentView: React.FC<Props>  = ({
    children,
    style
}) => {

    return Platform.OS === "android" ? (
        <ScrollView contentContainerStyle={style} >
          {children}
        </ScrollView>
      ) : (
        <KeyboardAvoidingView style={{ flex: 1 }} behavior="padding">
          <ScrollView contentContainerStyle={style} >
            {children}
          </ScrollView>
        </KeyboardAvoidingView>
      );
};

export default KeyboardAdjustmentView;



