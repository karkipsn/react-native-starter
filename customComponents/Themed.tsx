/**
 * Learn more about Light and Dark modes:
 * https://docs.expo.io/guides/color-schemes/
 */

import * as React from 'react';
import Colors from '../constants/Colors';
import useColorScheme from '../hooks/useColorScheme';
import { ThemeProps, TextProps, ViewProps} from './types'

import { Text as DefaultText, 
  View as DefaultView,
  Button as DefaultButton 
} from 'react-native';

export function useThemeColor(
  props: { light?: string; dark?: string },
  colorName: keyof typeof Colors.light & keyof typeof Colors.dark
) {
  const theme = useColorScheme();
  const colorFromProps = props[theme];

  if (colorFromProps) {
    return colorFromProps;
  } else {
    return Colors[theme][colorName];
  }
}

// Adding theme color in the props along with other props as array.
export const Text = (props: TextProps) => {

  const { style, lightColor, darkColor, ...otherProps } = props;
  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');

  return <DefaultText style={[{ color }, style]} {...otherProps} />;
}


// Adding backgroundcolor in the Style in the props along with other props as array.
export const View = (props: ViewProps) => {

  const { style, lightColor, darkColor, ...otherProps } = props;
  const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, 'background');

  return <DefaultView style={[{ backgroundColor }, style]} {...otherProps} />;
}




