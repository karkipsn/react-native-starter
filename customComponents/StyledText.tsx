import * as React from 'react';
import { Text } from './Themed';
import {  TextProps } from './types';
import { StyleSheet } from 'react-native';


// Text with Mono Styled color.
export const MonoText = (props: TextProps) => {
  return <Text {...props} style={[props.style, styles.monoText]} />;
}

// Bold text with Style of Red Color.
export const BoldText = (props: TextProps) => {
  return <Text {...props} style={[props.style, styles.boldText]} />;
}

// Custom RoundLabelText by inheriting textprops
export const RoundLabelText = (props: TextProps) =>{
  return <Text {...props} style={[props.style, styles.roundText]}  />
}

const styles = StyleSheet.create ({
  boldText: {
     padding: 4, 
     fontWeight: 'bold', 
     backgroundColor: 'red', 
     borderRadius: 40, 
     color: 'white' 
    },
  monoText: {
    fontFamily: 'space-mono', 
    borderRadius: 40
  },
  roundText: {
    backgroundColor: 'blue', 
    borderRadius: 12, 
    shadowColor: 'gray', 
    shadowRadius: 2,
  }

})
