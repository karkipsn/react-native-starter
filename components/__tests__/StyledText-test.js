import * as React from 'react';
import renderer from 'react-test-renderer';

import { MonoText, BoldText, RoundLabelText } from '../../customComponents/StyledText';

it(`Text renders correctly`, () => {

  const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON();
  expect(tree).toMatchSnapshot();

  const boldTree = renderer.create(<BoldText > Bold Text</BoldText>).toJSON();
  expect(boldTree).toMatchSnapshot();

  const roundText = renderer.create(<RoundLabelText>Round Text</RoundLabelText>).toJSON();
  expect(roundText).toMatchSnapshot();
});
