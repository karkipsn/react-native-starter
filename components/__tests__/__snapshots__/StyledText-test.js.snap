// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
      },
      Array [
        undefined,
        Object {
          "borderRadius": 40,
          "fontFamily": "space-mono",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`Text renders correctly 2`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
      },
      Array [
        undefined,
        Object {
          "backgroundColor": "red",
          "borderRadius": 40,
          "color": "white",
          "fontWeight": "bold",
          "padding": 4,
        },
      ],
    ]
  }
>
   Bold Text
</Text>
`;

exports[`Text renders correctly 3`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
      },
      Array [
        undefined,
        Object {
          "backgroundColor": "blue",
          "borderRadius": 12,
          "shadowColor": "gray",
          "shadowRadius": 2,
        },
      ],
    ]
  }
>
  Round Text
</Text>
`;
